runtime/src/main/java/com/vaadin/quarkus/QuarkusVaadinServletService.java

```diff
@@ -124,13 +144,53 @@ private void fireCdiDestroyEvent(ServiceDestroyEvent event) {
             // During application shutdown on TomEE 7,
             // beans are lost at this point.
             // Does not throw an exception, but catch anything just to be sure.
-            LoggerFactory.getLogger(QuarkusVaadinServletService.class)
-                    .warn("Error at destroy event distribution with CDI.", e);
+            getLogger().warn("Error at destroy event distribution with CDI.",
+                    e);
         }
     }
 
+    private void addUIListeners(UI ui) {
+        ui.addAfterNavigationListener(uiEventListener);
+        ui.addBeforeLeaveListener(uiEventListener);
+        ui.addBeforeEnterListener(uiEventListener);
+        ui.addPollListener(uiEventListener);
+    }
+
     private BeanManager getBeanManager() {
         return beanManager;
     }
 
+    private static Logger getLogger() {
+        return LoggerFactory.getLogger(QuarkusVaadinServletService.class);
+    }
+
+    /**
+     * Static listener class, to avoid registering the whole service instance.
```

Comment:
```suggestion
     * Inner listener class, to avoid registering the whole service instance.
```

Reply:
The comment with the class is copied from CDI but then the code is modified.

I made the class non static because it was convenient to reuse the existing methods.
But now the comments says WHY it was static. Even though I don't see any danger to keep implicit reference to the service instance it this listener I agree that `static` is slightly better than inner class.

So I rewrote the code and the comment says exactly what it says.